IN subquery; case expression (#1377)
cap-js/docs: java/working-with-cql/query-api.md

      .where(e -> e.get("name").substring(2).eq("ter"));
    ```

> #### Case-When-Then Expressions
> 
> Use a case expression to compute a value based on the evaluation of connditions. The following query converts a person's numeric age value to a textual ageTxt representation:
> 
> ```java
> Select.from(PERSONS).columns(
>   p -> p.name(),
>   p -> p.when(p.age().lt(25)).then("young")
>         .when(p.age().ge(75)).then("elderly")
>         .orElse("adult").as("ageTxt").type(CdsBaseType.String));
> ```
> 
> ::: tip
> You should specify the result type so that the runtime case respect it accordingly.
> :::

Reviewer: ```suggestion
#### Case-When-Then Expressions

Use a case expression to compute a value based on the evaluation of conditions. The following query converts the stock of Books into a textual representation as 'stockLevel':

```java
Select.from(BOOKS).columns(
  b -> b.title(),
  b -> b.when(b.stock().lt(10)).then("low")
        .when(b.stock().gt(100)).then("high")
        .orElse("medium").as("stockLevel").type(CdsBaseType.STRING));
```

::: tip
Specify the result type so that the runtime case respects it accordingly.
:::
```

> **Note:** Chaining `$outer` in nested subqueries is not supported.

### `IN` Subquery

An `IN` subquery is used to test if an element (or tuple of elements) of an outer query is contained in the result of a subquery. You can use an `IN` subquery in fluent style as well as in fluent style:

```java
// fluent style
Select.from(AUTHORS).where(author -> author.name().in(
    Select.from(ASTRONAUTS).columns(astro -> astro.name())
));
```

> or in tree style. In this example we check wether the tuple (`firstName`, `lastName`) is contained in the result of the subquery:

Reviewer: ```suggestion
or in tree style. In this example we check whether the tuple (`firstName`, `lastName`) is contained in the result of the subquery:
```
